hadoop-tools/hadoop-azure/src/test/java/org/apache/hadoop/fs/azurebfs/services/ITestBlobOperation.java

```diff
@@ -0,0 +1,229 @@
+
+        // Validate the content by comparing the md5 computed and the value obtained from server
+        Assertions.assertThat(op.getResult().getResponseHeader(CONTENT_MD5))
+                .describedAs("The content md5 value is not correct")
+                .isEqualTo(ContentMD5);
+        Assertions.assertThat(op.getResult().getStatusCode())
+                .describedAs("The creation failed")
+                .isEqualTo(HTTP_CREATED);
+    }
+
+    /**
+     * Verify that parallel creation of blob will lead to error when overwrite is false.
+     * @throws Exception
+     */
+    @Test
+    public void testParallelCreateBlob() throws Exception {
+        final AzureBlobFileSystem fs = getFileSystem();
+        final Configuration configuration = new Configuration();
+        configuration.addResource(TEST_CONFIGURATION_FILE_NAME);
+        AbfsClient abfsClient = getClient(fs);
+
+        AbfsConfiguration abfsConfiguration = new AbfsConfiguration(configuration,
+                configuration.get(FS_AZURE_ABFS_ACCOUNT_NAME));
+
+        // Gets the client.
+        AbfsClient testClient = Mockito.spy(TestAbfsClient.createTestClientFromCurrentContext(
+                abfsClient,
+                abfsConfiguration));
+
+        byte[] buffer = getRandomBytesArray(5);
+
+        // Create a test container to upload the data.
+        Path testPath = path(TEST_PATH);
+        String finalTestPath = testPath.toString().substring(testPath.toString().lastIndexOf("/"));
+
+        // Creates a list of request headers.
+        final List<AbfsHttpHeader> requestHeaders = TestAbfsClient.getTestRequestHeaders(testClient);
+        requestHeaders.add(new AbfsHttpHeader(CONTENT_LENGTH, String.valueOf(buffer.length)));
+        requestHeaders.add(new AbfsHttpHeader(X_MS_BLOB_TYPE, BLOCK_BLOB_TYPE));
+        // Updates the query parameters.
+        final AbfsUriQueryBuilder abfsUriQueryBuilder = testClient.createDefaultUriQueryBuilder();
+
+        TracingContext tracingContext = Mockito.spy(new TracingContext("abcd",
+                "abcde", FSOperationType.CREATE,
+                TracingHeaderFormat.ALL_ID_FORMAT, null));
+
+        // Create an ExecutorService with 4 threads
+        ExecutorService executorService = Executors.newFixedThreadPool(5);
+        List<Future<?>> futures = new ArrayList<>();
+        for (int i = 0; i < 5; i++) {
+            futures.add(executorService.submit(() -> {
+                try {
+                    AbfsRestOperation op = testClient.createPathBlob(testPath.toUri().getPath(),
+                            true, false, null, null, tracingContext);
+                    if (op.getResult().getStatusCode() == HTTP_CONFLICT) {
+                        throw new IOException("BlobAlreadyExists");
+                    }
+                } catch (IOException e) {
+                    throw new RuntimeException(e); // pass the exception through as-is
+                }
+            }));
+        }
```

Comment:
1. This need not enforce parallel hits.
2. If two request gets parallelize such that two request go to 2 diff machine at same time. Can server maintain isolation. Can two blob get created?
3. test requirement can be achieved from sequential run as well. Why parallelism is required.